util/sh/lib/banners.sh

```diff
@@ -0,0 +1,20 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+# shellcheck shell=bash
+
+add_license_banner() {
+  local outfile="$1"
+  local license_banner="# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0"
+  echo -e "$license_banner\n$(cat $outfile)" > "$outfile"
+}
+
+add_autogen_banner() {
+  local outfile="$1"
+  local cmd="$2"
+  local autogen_banner="# THIS FILE HAS BEEN GENERATED, DO NOT EDIT MANUALLY. COMMAND:
+# $cmd"
+  echo -e "$autogen_banner\n$(cat $outfile)" > "$outfile"
```

Comment:
Might this exceed the kernel's command-line buffer of 128KiB? (because the subshell-substitution effectively brings `outfile` into the `echo` string)

Reply:
fixed.